### ISSUE TITLE
Windows display scaling ignored by default (x64 build) in v6.2.7

### ISSUE THREAD

**Issue by CodeMarc11** [FrankBijnen/ExifToolGui#216]
v6.2.7 has improved wrt to Windows display scaling, but it still need parameters to scale correctly.

- With no parameters, the window comes out too small (no scaling).
- With "/scale" the UI elements and fonts are the correct size (scaled), although the window itself is still a small size and _menus_ still use a small font (not scaled). Of course, I can resize the window but that won't fix the menus (I did not resize the window though, in order to illustrate the differences).
- With "/highdpi=unaware" or "/highdpi=unaware /scale" everything is the correct size (scaled): UI elements, menu, and window size. However, text looks "soft", as if it was rendered small (unscaled) and then scaled up to the correct size, instead of being rendered at the correct scaled size. Note that with "/scale" the text (except for menus) look like they're rendered at the correct scaled size. This is entirely usable though, just a bit weird. I've seen the same thing with other apps, notably installers that use a dialog with a gray background.

This rest of this post will be long, as I attached images to illustrate the effects of the various parameters. This is all on Windows 11 64-bit, with display scaling 150%, running ExifToolGUI x64 build.

Window with no parameters (too small, not scaled):
![Window no parameters](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/65fb5227-8dd1-49c7-9b11-da4ce102b139)

Window with "/scale":
![Window with scale](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/bb07f848-5c28-4fc2-a258-d666ffe0dc6a)

Window with "/highdpi=unaware":
![Window with highdpi unaware](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/cb90ac23-bb74-4d50-8694-ccd9d9d053e5)

Window with both parameters: "/highdpi=unaware /scale":
![Window with highdpi unaware and scale](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/9b4d09a5-a287-48ff-893d-b5cf371df935)

I created an image comparing the effect of the various parameters to make it easier to see the differences, especially the font rendering. Warning: the image is large (1000x1700)! Make sure to view it at true size (i.e. 100%), or else viewing will add its own artifacts. Note that if you have display scaling set to anything besides 100%, the _browser itself_ will scale the image! Best to save it and use a real image viewer with a zoom control, to keep it at 100%, 1:1 pixel.
![Comparison](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/e14d55c3-d447-4392-96ee-18010d21e827)

**Comment by FrankBijnen** [comment]
@CodeMarc11 Thanks for your detailed tests!

I have to read it a few times more and think about it. Will keep you posted

Frank

**Comment by CodeMarc11** [comment]
You're welcome. ☺️ I remember you don't have a 4K display to see the effect of scaling (and I think you're not using Win11 either), so figured I'd go overboard and post a bunch of screenshots.

What version of Windows are you able to test with? I'm typing this on the Win11 machine with 150% scaling, but also have a Win10 machine with no scaling (100% I guess?). Though I haven't tried ExifToolGUI on it, expect it will be fine on there.

**Comment by CodeMarc11** [comment]
Some more info, in case it's useful...

I just saw the last posts on the exiftool forum, and tried the scaling options in the Properties dialog:
![Run exe - Settings dialog](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/5253f6e4-b1eb-49d5-8c1e-5dc1b656d023)
This is Properties directly on the executable, not a shortcut, so it's running with no parameters.

Under Program DPI, checking the box for "Use this setting..." made no difference, with either dropdown choice under "Use the DPI..." (app has small window).

However, if I went under "High DPI scaling override" and checked "Override high DPI scaling behavior" and changed the dropdown to "System" or "System (Enhanced)", then I got the same window as using a shortcut with parameter "/highdpi=unaware".
![Run exe with Scaling override System](https://github.com/FrankBijnen/ExifToolGui/assets/38742926/0d4190f8-24ea-4522-8795-35b5203cf13a)
With the dropdown default "Application", it made no difference (small window). Note that the fonts (and fine details in the window) are still "soft" as if upscaled. "System (Enhanced)" was still soft, just like "System".

No rush, just more to think about. ☺️
